src/agent/structures/events.ts

```diff
@@ -93,15 +98,42 @@ export interface Image {
 }
 
 export interface CustomEvent extends BaseEvent {
+  type: "custom";
   custom_id?: string;
   author_id: string;
   content?: object;
 }
 
 export interface SystemMessage extends BaseEvent {
+  type: "system_message";
   system_message_type: string;
   text?: string;
   text_vars?: object;
 }
 
 export type Event = File | FilledForm | Message | RichMessage | CustomEvent | SystemMessage;
+
+type ResponseOnlyFields =
+  | "id"
+  | "created_at"
+  | "visibility"
```

Comment:
isn't this field also in req?

Reply:
`visibility` field is always present in the response, but it is optional in the request - [docs](https://developers.livechat.com/docs/messaging/agent-chat-api/data-structures)

I don't think there is a better name than `ResponseOnlyFields`, even though it's only partially true for `visibility`